extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -221,28 +228,105 @@ void recordBouncyCastleProviders(SecurityProviderRecorder recorder,
     }
 
     @BuildStep
-    void addBouncyCastleProvidersToNativeImage(BuildProducer<NativeImageSecurityProviderBuildItem> additionalProviders,
+    void addBouncyCastleProvidersToNativeImage(
+            BuildProducer<GeneratedNativeImageClassBuildItem> nativeImageClass,
+            BuildProducer<NativeImageSecurityProviderBuildItem> additionalProviders,
             List<BouncyCastleProviderBuildItem> bouncyCastleProviders,
             List<BouncyCastleJsseProviderBuildItem> bouncyCastleJsseProviders) {
         Optional<BouncyCastleJsseProviderBuildItem> bouncyCastleJsseProvider = getOne(bouncyCastleJsseProviders);
         if (bouncyCastleJsseProvider.isPresent()) {
             additionalProviders.produce(
                     new NativeImageSecurityProviderBuildItem(SecurityProviderUtils.BOUNCYCASTLE_JSSE_PROVIDER_CLASS_NAME));
-            final String providerName = bouncyCastleJsseProvider.get().isInFipsMode()
-                    ? SecurityProviderUtils.BOUNCYCASTLE_FIPS_PROVIDER_CLASS_NAME
-                    : SecurityProviderUtils.BOUNCYCASTLE_PROVIDER_CLASS_NAME;
-            additionalProviders.produce(new NativeImageSecurityProviderBuildItem(providerName));
+
+            ClassCreator file = createBouncyCastleFeatureCreator(nativeImageClass);
+            MethodCreator beforeAn = createBeforeAnalysisMethod(file);
+            TryBlock overallCatch = beforeAn.tryBlock();
```

Comment:
This is common code in both `if` and `else` blocks, I would move it before the if-statement to avoid code duplication.

This would also remove the need for defining the `createBeforeAnalysisMethod` method.

Reply:
Hi @zakkak In both cases (here and in the other one you comment at below) it is conditional, 
```
Optional<BouncyCastleJsseProviderBuildItem> bouncyCastleJsseProvider = getOne(bouncyCastleJsseProviders);
if (bouncyCastleJsseProvider.isPresent()) {
           
            ClassCreator file = createBouncyCastleFeatureCreator(nativeImageClass);
            MethodCreator beforeAn = createBeforeAnalysisMethod(file);
            TryBlock overallCatch = beforeAn.tryBlock();

            if (!bouncyCastleJsseProvider.get().isInFipsMode()) {
                //...
            } else {
                //...
            }
            completeBouncyCastleFeature(file, beforeAn, overallCatch);

        } else {
            Optional<BouncyCastleProviderBuildItem> bouncyCastleProvider = getOne(bouncyCastleProviders);
            if (bouncyCastleProvider.isPresent()) {
                
                ClassCreator file = createBouncyCastleFeatureCreator(nativeImageClass);
                MethodCreator beforeAn = createBeforeAnalysisMethod(file);
                TryBlock overallCatch = beforeAn.tryBlock();

                //...
                completeBouncyCastleFeature(file, beforeAn, overallCatch);
            }
        }
```

See, I can't move it the class creator code above the top `if` (and closing it outside it).
Am I missing something ?
Thanks

Reply:
I.e, there is a branch which allows for no BC or BC JSSE providers found at all

Reply:
@zakkak I'll collapse both ifs

Reply:
@zakkak I've done a small refactoring, hope it looks cleaner now, thanks

Reply:
You are right, I missed the if-statement in the `else` body... Sorry for the noise

Reply:
@zakkak Np at all, I still followed up with a minor refactoring, to avoid nested if inside the outer else which was not too cool, so thanks for your suggestion.